### DISCUSSION TITLE
Google sign in Error

### DISCUSSION THREAD

**Discussion by ajay-k07** [serverpod/serverpod#505] [Q&A] [answered]
Hi i was trying to create a google sign in tutorial i was able to complete for web and ios, but for android im facing issues 

![Screenshot 2022-12-20 at 4 21 10 PM](https://user-images.githubusercontent.com/37687348/208724104-c444b942-f3bf-44c1-8af8-1eb7b5e6e089.png)

Code snipped i used 

```
SignInWithGoogleButton(
                    caller: client.modules.auth,
                    clientId:
                        'xxxxxx.apps.googleusercontent.com',
                    serverClientId:
                        'xxxxxxx.apps.googleusercontent.com',
                    redirectUri: Uri.parse('http://localhost:53055'),
                    onSignedIn: () => setState(() {}),
                  ),
```

```
SignInWithGoogleButton(
                    caller: client.modules.auth,
                    clientId:
                        'xxxxxx.apps.googleusercontent.com',
                    redirectUri: Uri.parse('http://localhost:53055'),
                    onSignedIn: () => setState(() {}),
                  ),
```

when i comment out the following code in   serverpod_auth_google_flutter/lib/src/auth.dart signInWithGoogle it was working fine 

```
    // Authentication with server is complete, we can sign out from Google locally
    if (kDebugMode) print('serverpod_auth_google: Signing out from google');
  
      await googleSignIn.signOut();


    //if (kDebugMode) await googleSignIn.disconnect();

    if (kDebugMode) {
      print(
        'serverpod_auth_google: Authentication was successful. Saved credentials to SessionManager',
      );
    }
```

**Comment by yahu1031** [comment, marked as answer]
@ajay-k07 Ideally, You need to `disconnect()` first and then need `signOut()`. Try it this way and check if that works.

**Reply by ajay-k07** [reply]
Ok i will do the same. If it's the proper way can i fix that and create an PR

**Reply by yahu1031** [reply]
Check out the diff here for [`disconnect()`](https://pub.dev/documentation/google_sign_in/latest/google_sign_in/GoogleSignIn/disconnect.html) and [`signOut()`](https://pub.dev/documentation/google_sign_in/latest/google_sign_in/GoogleSignIn/signOut.html)

**Reply by yahu1031** [reply]
`disconnect` - will disconnect the user from the app, such that you need to go through the OAuth flow again if you wanna log in.

`signOut` - The account instances will be saved but the account will be marked as signed out. So when you click on sign in, It will automatically log the user in.

**Reply by yahu1031** [reply]
@ajay-k07 If this works, please mark it as answered.

**Reply by ajay-k07** [reply]
Sure will do that once i try that will do by tonight
